platform/openide.execution/src/org/openide/execution/ExecutionEngine.java

```diff
@@ -127,13 +127,14 @@ public ExecutorTask execute(String name, Runnable run, InputOutput io) {
             private RequestProcessor.Task task;
             private int resultValue;
             private final String name;
-            private InputOutput io;
+            private final InputOutput io;
```

Comment:
NPE due to memory model!? E.g. it is set, but other thread doesn't see the value? Interesting.

Reply:
Note. The field [was simply not assigned at all](https://github.com/apache/netbeans/pull/2982/files#diff-164cf44c2b7deea8554b6c153dbac61fa62df18741fe3e6fc363d8d4a576d5f9R137). Just added `final` since the constructor got the delegate, and there were no attempts to change / invalidate it. But yes, could eventually happen even that case.